src/Core/src/HttpClient/AwsHttpClientFactory.php

```diff
@@ -12,11 +12,23 @@
  */
 class AwsHttpClientFactory
 {
+    public static function createClient(): HttpClientInterface
+    {
+        return new CurlHttpClient([
+            'extra' => [
+                'curl' => [
+                    \CURLOPT_SSLVERSION => \CURL_SSLVERSION_TLSv1_2,
+                ],
+            ],
+        ]);
```

Comment:
I'm not sure we should force people to use an extension
what about making curl a soft dependency with something like (not tested):

```suggestion
        $options = [];
        if (\extension_loaded('curl')) {
           $options = [
                'extra' => [
                    'curl' => [
                        \CURLOPT_SSLVERSION => \CURL_SSLVERSION_TLSv1_2,
                    ],
                ],
            ];
        }

        return HttpClient::create($options);
        
```

What do you think @nicolas-grekas

Reply:
If we do this, then it doesn't enforce TLS 1.2. I'd rather have it fail loud if curl is not installed, than silently fallback. If we were to have it silently fallback, this would not solve the issue I created the PR to solve.

Reply:
> I'm not sure we should force people to use an extension

If people are not using curl, they can explicitly opt out by passing a client themselves. That seems like a more secure and robust default than silently not being async or not requiring TLS 1.2.